ttnn/cpp/ttnn/operations/data_movement/repeat/repeat.cpp

```diff
@@ -36,11 +47,26 @@ ttnn::Tensor RepeatOperation::invoke(
                         (input_tensor.get_legacy_shape()[dim] * input_tensor.element_size()) % input_tensor.buffer()->alignment() == 0,
                         "Current repeat implementation requires aligned last dim when repeating on last dim");
                 }
-                output = operation::run_without_autoformat(RepeatDeviceOperation{dim, repeat_dims[dim], memory_config}, {output}).at(0);
+                auto outputs = operation::run_without_autoformat(RepeatDeviceOperation{dim, repeat_dims[dim], memory_config}, {output});
+                TT_FATAL(outputs.size() == 1, "ttnn.repeat: expected 1 output tensor from run_without_autoformat, but got {}", outputs.size());
+                output = outputs[0];
             }
             return {output};
-        }, {input_tensor}, output_tensors);
-    return output_tensors.at(0);
+        }, {formatted_input_tensor.has_value() ? *formatted_input_tensor : input_tensor}, output_tensors);
+    TT_FATAL(output_tensors.size() == 1, "ttnn.repeat: expected 1 output tensor, but got {}", output_tensors.size());
+    if (formatted_input_tensor.has_value()) {
+        auto zero_indices = std::vector<uint32_t>(input_tensor.get_legacy_shape().rank(), 0);
```

Comment:
I think it will be better to request shape once and then work with it. 
```
auto shape = input_tensor.get_logical_shape();
```

every time you call `without_padding` you make a copy of LegacyShape, let's avoid redundant copies (especially in for loops).

Reply:
Done :)